### DISCUSSION TITLE
Updating submodules with tags and storing submodule revision in textfile

### DISCUSSION THREAD

**Discussion by defaultuserr** [renovatebot/renovate#26027] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.50.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

Im labeling it as "Question" since I do not know if this is already possible with renovate. 
We already successfully manage many projects with renovate gitlab runner for most of our usecases. There is one usecase i would like to ask if it is possible right now to be managed by renovate.

Lets say we have  a project we want to include as a submodule, `submodule-project` and a project which wants to include the summodule `include-submodule-project`.
Including the `submodule-project`  in `include-submodule-project` and getting it updated via renovate works like a charm. We receive a MR in `include-submodule-project` which the updated digest.

Lets also say we also have a text file in `include-submodule-project`  which contains the version of the submodule, lets call it `version.txt`. Every time renovate wants to update the submodule in `include-submodule-project`, the version in `version.txt` should also be updated accordingly. 

Question 1: Is it possble to let renovate run only on tags in the submodule and not getting an update on every push to the configured branch of the submodule? 
Question2 : The versions in `version.txt` should be tags of the submodule, not the hash. Is this somehow possible? I found different use cases for customManagers, but not this specific use case. 
My idea would be to have something in my config.js like that, but I did not manage to find anyone doing something similar and therefore im not sure if this can possibly work.
```
"customManagers": [
    {
      "customType": "regex",
      "datasourceTemplate": "git-refs",
      "fileMatch": [
          "version\\.txt$"
      ],
      "matchStrings": [
        "dependency-from-submodule[(?<currentValue>.*)]"
      ]
    }
  ]
```

If something is not clear feel free to ask.
Thank you for reading and answering my question :)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by anarcat** [comment]
At least *part* of this question is covered by https://github.com/renovatebot/renovate/discussions/24890, but that doesn't have an answer either, as far as i can tell.

**Comment by defaultuserr** [comment]
I kind of solved it this way:
I have a custom manager with dataSource gitlab-tags and a submodule manager. I group them via a PackagaManager via MatchManagers.
This way, i only receive MR with the updated files and the updated submodule. This kind of solves it, but is it not what i would prefer.
About the only git tag trigger, i added a downtream pipeline trigger from the submodule project to the renovate schedule. 

With this solution, the renovate must not run on a schedule since inconsistency between the submodule hash and the versions in the text file can occur since they are 2 different managers. 

This solution works at least for us.

**Reply by anarcat** [reply]
>  I have a custom manager with dataSource gitlab-tags and a submodule manager. I group them via a PackagaManager via MatchManagers.

Oh that! What does that look like in JSON, concretely?
